### PR TITLE
server, config: remove all remaining unwraps

### PR DESCRIPTION
**Pull request by flouthoc**
Removes some last remaining bits of unwrap usage

Edit: Although all the necessary unwrap were removed before but this makes sure we don't have any unwrap traces at all.

This completely closes the issue: https://github.com/containers/aardvark-dns/issues/59

### CODE REVIEW

File: src/server/serve.rs [hunk 1]
```diff
@@ -271,17 +271,22 @@ async fn start_dns_server(
...
+        if let Ok(conn) = UdpClientConnection::with_timeout(address, Duration::from_millis(5)) {
+            // and then create the Client
+            let client = SyncClient::new(conn);
+            // server will be killed by last request
+            if let Ok(name) = Name::from_str("anything.") {
+                match client.query(&name, DNSClass::IN, RecordType::A) {
+                    _ => {}
+                }
+            }
+        }
```
Comment (Luap99) [inline src/server/serve.rs:290]: I do not understand why we need this? I think send_broadcast will already notify the listener threads

Comment (flouthoc) [reply src/server/serve.rs:290]: @Luap99 Yes this should not be needed anymore, I already added a marker here in the past: https://github.com/containers/aardvark-dns/blob/main/src/server/serve.rs#L191 but just want to test and do that in a different PR so its easier to review and revert.

**Comment by flouthoc** [comment]
@Luap99 @baude Good to go. I think

**Comment by Luap99** [comment]
I will take a look at the slirp timeout flake later

**Pull request merged**